### ISSUE TITLE
glowroot 0.13.6 release does not work with AdoptOpenJDK 15

### ISSUE THREAD

**Issue by mschaaf** [glowroot/glowroot#765]
Having an official release compiled for JDK 15 would be nice. Thanks.

Exception is:

```
Caused by: java.lang.VerifyError: Stack map does not match the one at exception handler 150 
Exception Details: 
  Location: 
    com/sun/proxy/$Proxy149.getConnection(Ljava/lang/String;Ljava/lang/String;)Ljava/sql/Connection; @150: iload_3 
  Reason: 
    Type 'java/lang/Throwable' (current frame, locals[1]) is not assignable to 'java/lang/String' (stack map, locals[1]) 
  Current Frame: 
    bci: @115 
    flags: { } 
    locals: { 'com/sun/proxy/$Proxy149', 'java/lang/Throwable', 'java/lang/String', integer, 'org/glowroot/agent/bytecode/api/ThreadContextThreadLocal$Holder', 'org/glowroot/agent/bytecode/api/ThreadContextPlus', integer, 'java/lang/Object' } 
    stack: { 'java/lang/Throwable' } 
  Stackmap Frame: 
    bci: @150 
    flags: { } 
    locals: { 'com/sun/proxy/$Proxy149', 'java/lang/String', 'java/lang/String', integer, 'org/glowroot/agent/bytecode/api/ThreadContextThreadLocal$Holder', 'org/glowroot/agent/bytecode/api/ThreadContextPlus', integer, 'java/lang/Object' } 
    stack: { 'java/lang/Throwable' } 
  Bytecode: 
    0000000: 033e 013a 0401 3a05 0236 0601 3a07 b800 
    0000010: e599 0034 1904 c700 08b8 00eb 3a04 1904 
    0000020: b600 f359 3a05 c600 1f19 05b9 00f6 0100 
    0000030: 5936 0612 f79f 0010 1905 12f7 b900 fb02 
    0000040: 0004 a700 0403 3e1d 9900 0a19 05b8 00ff 
    0000050: 3a07 2ab4 00b2 2ab2 010e 05bd 00ca 5903 
    0000060: 2b53 5904 2c53 b900 ba04 00c0 0103 a700 
    0000070: 0ebf 4cbb 00c1 592b b700 c4bf 1d99 0009 
    0000080: 5919 07b8 0107 1506 029f 000c 1905 1506 
    0000090: b900 fb02 00b0 1d99 0009 5919 07b8 010b 
    00000a0: 1506 029f 000c 1905 1506 b900 fb02 00bf 
    00000b0:                                         
  Exception Handler Table: 
    bci [82, 113] => handler: 113 
    bci [82, 113] => handler: 113 
    bci [82, 113] => handler: 113 
    bci [82, 113] => handler: 114 
    bci [82, 124] => handler: 150 
  Stackmap Table: 
    full_frame(@30,{Object[#2],Object[#214],Object[#214],Integer,Object[#237],Object[#239],Integer,Object[#202]},{}) 
    same_frame(@69) 
    same_locals_1_stack_item_frame(@70,Integer) 
    same_frame(@82) 
    same_locals_1_stack_item_frame(@113,Object[#174]) 
    same_locals_1_stack_item_frame(@114,Object[#174]) 
    same_locals_1_stack_item_frame(@124,Object[#259]) 
    same_locals_1_stack_item_frame(@134,Object[#259]) 
    same_locals_1_stack_item_frame(@149,Object[#259]) 
    same_locals_1_stack_item_frame(@150,Object[#174]) 
    same_locals_1_stack_item_frame(@160,Object[#174]) 
    same_locals_1_stack_item_frame(@175,Object[#174]) 
 
        at java.lang.Class.getDeclaredConstructors0(Native Method) ~[?:?] 
        at java.lang.Class.privateGetDeclaredConstructors(Class.java:3296) ~[?:?] 
        at java.lang.Class.getConstructor0(Class.java:3501) ~[?:?] 
        at java.lang.Class.getConstructor(Class.java:2244) ~[?:?] 
        at java.lang.reflect.Proxy$ProxyBuilder.build(Proxy.java:651) ~[?:?] 
        at java.lang.reflect.Proxy.lambda$getProxyConstructor$1(Proxy.java:428) ~[?:?] 
        at jdk.internal.loader.AbstractClassLoaderValue$Memoizer.get(AbstractClassLoaderValue.java:329) ~[?:?] 
        at jdk.internal.loader.AbstractClassLoaderValue.computeIfAbsent(AbstractClassLoaderValue.java:205) ~[?:?] 
        at java.lang.reflect.Proxy.getProxyConstructor(Proxy.java:426) ~[?:?] 
        at java.lang.reflect.Proxy.getProxyClass(Proxy.java:386) ~[?:?] 
        at org.springframework.util.ClassUtils.createCompositeInterface(ClassUtils.java:784) ~[spring-core-5.2.12.RELEASE.jar!/:5.2.12.RELEASE] 
        at org.springframework.aop.aspectj.AspectJExpressionPointcut.getTargetShadowMatch(AspectJExpressionPointcut.java:437) ~[spring-aop-5.2.12.RELEASE.jar!/:5.2.12.RELEASE] 
        at org.springframework.aop.aspectj.AspectJExpressionPointcut.matches(AspectJExpressionPointcut.java:295) ~[spring-aop-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
        at org.springframework.aop.support.AopUtils.canApply(AopUtils.java:251) ~[spring-aop-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
        at org.springframework.aop.support.AopUtils.canApply(AopUtils.java:289) ~[spring-aop-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
        at org.springframework.aop.support.AopUtils.findAdvisorsThatCanApply(AopUtils.java:321) ~[spring-aop-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]

```

**Comment by iam8139** [comment]
Hii @mschaaf, can you post your answer if you have resolved this issue !!

**Comment by stefpiatek** [comment]
I think this is the same issue as #820 #738 #721 and #711 . Bytecode validation thowing an exception, presumably because of conflicting versions of dependencies in the monitored project and glowroot itself? That's the best that I've found but that level is beyond me.

**Comment by mschaaf** [comment]
Disabling bytecode verification on the JVM solved the issue for me.

**Comment by trask** [comment]
hi all, this is fixed now in the 0.14.0-beta.2 release

**Comment by stefpiatek** [comment]
Thanks @trask, tested the beta release along with an upgrade to JDK 17 and its all up and working 🎉
